### ISSUE TITLE
Bump required version of mypy-zope to 0.3.0

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10172]
|[<img alt="wiml's avatar" src="https://avatars.githubusercontent.com/u/156891?s=50" width="50" height="50">](https://github.com/wiml)| @wiml reported|
|-|-|
|Trac ID|trac#10172|
|Type|enhancement|
|Created|2021-04-08 20:42:39Z|

mypy-zope 0.3.0 adds support for using the ```@overload``` decorator in interfaces ( see https://github.com/Shoobx/mypy-zope/issues/46 ), which is needed in at least one place in Twisted.

<details><summary>Searchable metadata</summary>

```
trac-id__10172 10172
type__enhancement enhancement
reporter__wiml wiml
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__new new
resolution__None None
component__core core
keywords__None None
time__1617914559526143 1617914559526143
changetime__1618573297248737 1618573297248737
version__None None
owner__graingert graingert

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wiml's avatar" src="https://avatars.githubusercontent.com/u/156891?s=50" width="50" height="50">](https://github.com/wiml)|@wiml set owner to @graingert|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wiml's avatar" src="https://avatars.githubusercontent.com/u/156891?s=50" width="50" height="50">](https://github.com/wiml)<a name="note_2"></a>|@wiml commented|
|-|-|

ticket:10176 was a duplicate of this

**Comment by twisted-trac** [comment]
|[<img alt="graingert's avatar" src="https://avatars.githubusercontent.com/u/413772?s=50" width="50" height="50">](https://github.com/graingert)<a name="note_3"></a>|@graingert commented|
|-|-|

when upgrading mypy-zope we see new errors in the IReactorThreads implementation in ReactorBase:


```
16:51:52 <graingert> glyph: in mypy-zope==0.3.0 twisted.internet.base.ReactorBase.callInThread takes an `f: Callable[...]`
16:52:08 <graingert> but IReactorInThreads.callInThread takes a callable: Callable[...]
16:53:20 <graingert> really they should both be  callInThread(callable: Callable, \, *args: object, **kwargs)
16:53:27 <graingert> but we can't do that until py3.8
16:54:15 <graingert> what should reactor.callInThread(callable=lambda f: None, f=2) do?
16:54:24 <graingert>  /cc exarkun 
16:55:23 <graingert> the stdlib solution was to support only kwargs
16:58:18 <graingert> it's also possible to do:
16:58:18 <graingert> def callFromThread(*args: object, **kwargs: object):
16:58:18 <graingert>     f, *args = args
```

**Comment by twisted-trac** [comment]
|[<img alt="graingert's avatar" src="https://avatars.githubusercontent.com/u/413772?s=50" width="50" height="50">](https://github.com/graingert)<a name="note_4"></a>|@graingert commented|
|-|-|

https://github.com/twisted/twisted/pull/1585

**Comment by exarkun** [comment]
PR merged in 2eeaacbb960a3380f9d9b91e1f9197e22199c5ff
